docs: Add section about capacity planning (#4386)
fastify/fastify: docs/Guides/Recommendations.md

    timeoutSeconds: 3
    successThreshold: 1
    failureThreshold: 5

> ```

Reviewer: Are these single tickets on purpose?

> ## Capacity planning for production

Reviewer 2: This needs to pass a message of "do your own measurements".

Author: @mcollina Adjusted. Do you think that the emphasis is correct now?

    timeoutSeconds: 3
    successThreshold: 1
    failureThreshold: 5
```

> ## Capacity planning for production

Reviewer 3: For consistency in titles:

```suggestion
## Capacity Planning For Production
```

<a id="capacity"></a>

> In order to rightsize your fastify application deployment in production, 
> you are highly recommended to perform your own measurements against 
> different configurations of the production environment 
> (2 vCPU, 1 vCPU, fractional parts of vCPU).
> 

Reviewer 3: Given this is a standalone section, we shouldn't limit this language to "vCPU". The user could be running on actual hardware or their environment could use different terminology.

```suggestion
In order to rightsize your Fastify application deployment in production, 
you are highly recommended to perform your own measurements against 
different configurations of the production environment, which may
use real CPU cores, virtual CPU cores (vCPU), or even fractional
vCPU cores. We will use the term vCPU throughout this
recommendation to represent any CPU type.

```

You can use such tools as [k6](https://github.com/grafana/k6) 
or [autocannon](https://github.com/mcollina/autocannon) for conducting
the necessary performance tests.

That said, you may also consider the following as a rule of a thumb:

> * In order to have the lowest possible latency, 2 vCPU are recommended per app 
> instance (k8s pod). The second vCPU will mostly be used by the GC and libuv
> threadpool. This will minimize the latency for your users, as well as the
> memory usage, as the GC will be run more frequently. The main thread won't
> have to stop to let the GC run.

Reviewer 3: ```suggestion
* In order to have the lowest possible latency, 2 vCPU are recommended per app 
instance (e.g., a k8s pod). The second vCPU will mostly be used by the garbage collector (GC) and libuv
threadpool. This will minimize the latency for your users, as well as the
memory usage, as the GC will be run more frequently. Also, the main thread won't
have to stop to let the GC run.
```

    timeoutSeconds: 3
    successThreshold: 1
    failureThreshold: 5
```

## Capacity Planning For Production
<a id="capacity"></a>

In order to rightsize the production environment for your Fastify application, 
you are highly recommended to perform your own measurements against 
different configurations of the environment, which may
use real CPU cores, virtual CPU cores (vCPU), or even fractional
vCPU cores. We will use the term vCPU throughout this
recommendation to represent any CPU type.

You can use such tools as [k6](https://github.com/grafana/k6) 
or [autocannon](https://github.com/mcollina/autocannon) for conducting
the necessary performance tests.

That said, you may also consider the following as a rule of a thumb:

* In order to have the lowest possible latency, 2 vCPU are recommended per app 
instance (e.g., a k8s pod). The second vCPU will mostly be used by the 
garbage collector (GC) and libuv threadpool. This will minimize the latency 
for your users, as well as the memory usage, as the GC will be run more
frequently. Also, the main thread won't have to stop to let the GC run.

* In order to optimize for throughput (handling the largest possible amount of

> requests per second per vCPU), consider using smaller amount of vCPUs per app 

Reviewer 2: ```suggestion
requests per second per vCPU available), consider using smaller amount of vCPUs per app 
```
